src/RefreshOptions.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { WatchedSetting } from "./WatchedSetting";
+
+export const DefaultRefreshIntervalInMs = 30 * 1000;
+export const MinimumRefreshIntervalInMs = 1 * 1000;
+
+export interface RefreshOptions {
+    /**
+     * Specifies whether the provider should automatically refresh when the configuration is changed.
+     */
+    enabled: boolean;
+
+    /**
+     * Specifies the minimum time that must elapse before checking the server for any new changes.
+     * Default value is 30 seconds. Must be greater than 1 second.
+     * Any refresh operation triggered will not update the value for a key until after the interval.
+     */
+    refreshIntervalInMs?: number;

```

Comment:
refreshInterval is misleading. This is actually for backoff.

Reply:
try {
      await this.load(RequestType.Watch);
      this.#refreshTimer.reset();
  } catch (error) {
      // if refresh failed, backoff
      this.#refreshTimer.backoff();
      throw error;
  }
  If `load()` successfully each time we call `refresh()`, the refreshInterval will not work at all.

Reply:
The refresh option in dotnet provider provide a method to set cache expiration interval for watched key.
https://github.com/Azure/AppConfiguration-DotnetProvider/blob/main/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationRefreshOptions.cs#L58

The cache expiration time of watched key will be update through this method
https://github.com/Azure/AppConfiguration-DotnetProvider/blob/main/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs#L941

@avanigupta

Reply:
The python provider also uses this misleading name. @mrm9084

Reply:
It’s affected by the exponential backoff, indeed we should update the comments here to clarify the behavior. I myself think CacheExpires is more confusing, because here actually no cache is preserved. What it does is, it prevents sending requests within a specific period of time, as if there is a cache. So that’s why we use the name “refresh interval” instead.

The refresh interval works when there is no failure. In the first few lines of refresh(), it checks #refreshTimer.canRefresh() , which checks if refresh interval is passed.

Reply:
The "cache expiration" naming in .NET has caused confusion in the past and we are trying to move away from it ([tracking work item](https://github.com/Azure/AppConfiguration-DotnetProvider/issues/350)).
The naming `refreshIntervalInMs` is appropriate here because we are asking users to set the minimum time between server calls to check for changes. So this is the interval between 2 refresh operations.
Backoff is our internal concept that comes into play when we fail to read key-values from the server. Users shouldn't need to know the implementation details of our error handling strategies. So "backoff" would not be a good name to add to any user facing properties.

Reply:
Ok, then let's make no change here as explained.